packages/financial-templates-lib/src/logger/DiscordTicketTransport.ts

```diff
@@ -77,8 +93,9 @@ export class DiscordTicketTransport extends Transport {
         // Prepend the $ticket command and concatenate message title and content separated by newline.
         const message = `$ticket ${info.message}\n${removeAnchorTextFromLinks(info.mrkdwn)}`;
 
-        // Send the message.
-        await channel.send(message);
+        // Add the message to the queue and process it.
```

Comment:
There are weird edge cases like someone checking isFlushed while we're awaiting the channel fetch.

To avoid all the hard-to-reason-about async edge cases, why not just try to use isFlushed to signal is _any_ log call is ongoing?

The way I might approach this is:
1. Create a counter class variable.
2. Increments each time the log method is called.
3. Decrement each time the log method finishes (by error or success).
4. When a user tries to check if this has flushed, we just return true if the counter is not 0 (you can use a getter/setter _or_ make isFlushed a method, either way works).

Reply:
This totally makes sense. I made internal methods private and use log call counter to detect when the transport is flushed. Also created isFlushed getter to check when counter is back to 0